openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -123,6 +124,10 @@ def process_reference(self, context, name, namespace, options):
         attach_to_root = options.get("attach_to_root", True)
         group_name = options["group_name"]
 
+        # no group shall be created
+        if not attach_to_root:
+            group_name = namespace
```

Comment:
We can also filter `cmds.ls` to only ever returning a single node to begin with, to speed it up:
```python
group_name = cmds.ls(new_nodes, assemblies=True, long=True, head=1)[0]
```
So I'd likely recommend adding `head=1`.

Note that the above logic would fail if a loaded reference would have NO dag nodes whatsoever (e.g. a look).

So we might want to do:
```python
group_name = next(iter(cmds.ls(new_nodes, assemblies=True, long=True, head=1)), None)
```

Or if you prefer this readability wise:
```python
roots = cmds.ls(new_nodes, assemblies=True, long=True, head=1)
group_name = roots[0] if roots else None
```

Note that any logic that would touch `group_name` would still need to check whether it's not None.